presto-hive/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -2289,36 +2290,40 @@ public void undo(ExtendedHiveMetastore metastore)
 
     private static class AlterPartitionOperation
     {
-        private final Partition newPartition;
-        private final Partition oldPartition;
-        private boolean done;
+        private final PartitionWithStatistics newPartition;
+        private final PartitionWithStatistics oldPartition;
+        private boolean undo;
 
-        public AlterPartitionOperation(Partition newPartition, Partition oldPartition)
+        public AlterPartitionOperation(PartitionWithStatistics newPartition, PartitionWithStatistics oldPartition)
         {
             this.newPartition = requireNonNull(newPartition, "newPartition is null");
             this.oldPartition = requireNonNull(oldPartition, "oldPartition is null");
-            checkArgument(newPartition.getDatabaseName().equals(oldPartition.getDatabaseName()));
-            checkArgument(newPartition.getTableName().equals(oldPartition.getTableName()));
-            checkArgument(newPartition.getValues().equals(oldPartition.getValues()));
+            checkArgument(newPartition.getPartition().getDatabaseName().equals(oldPartition.getPartition().getDatabaseName()));
+            checkArgument(newPartition.getPartition().getTableName().equals(oldPartition.getPartition().getTableName()));
+            checkArgument(newPartition.getPartition().getValues().equals(oldPartition.getPartition().getValues()));
         }
 
         public String getDescription()
         {
-            return format("alter partition %s.%s %s", newPartition.getDatabaseName(), newPartition.getTableName(), newPartition.getValues());
+            return format(
+                    "alter partition %s.%s %s",
+                    newPartition.getPartition().getDatabaseName(),
+                    newPartition.getPartition().getTableName(),
+                    newPartition.getPartition().getValues());
         }
 
         public void run(ExtendedHiveMetastore metastore)
         {
-            metastore.alterPartition(newPartition.getDatabaseName(), newPartition.getTableName(), newPartition);
-            done = true;
+            undo = true;
+            metastore.alterPartition(newPartition.getPartition().getDatabaseName(), newPartition.getPartition().getTableName(), newPartition);
```

Comment:
Good catch. This is a pre-existing bug that can have visible effect before the stats change.